cmd/rpcdaemon/commands/parity_api.go

```diff
@@ -0,0 +1,68 @@
+package commands
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/common"
+	"github.com/ledgerwatch/erigon/common/hexutil"
+	"github.com/ledgerwatch/erigon/core/state"
+)
+
+// ParityAPI the interface for the parity_ RPC commands
+type ParityAPI interface {
+	ListStorageKeys(ctx context.Context, account common.Address, quantity int, offset *hexutil.Bytes) ([]hexutil.Bytes, error)
+}
+
+// ParityAPIImpl data structure to store things needed for parity_ commands
+type ParityAPIImpl struct {
+	db kv.RoDB
+}
+
+// NewParityAPIImpl returns ParityAPIImpl instance
+func NewParityAPIImpl(db kv.RoDB) *ParityAPIImpl {
+	return &ParityAPIImpl{
+		db: db,
+	}
+}
+
+// ListStorageKeys implements parity_listStorageKeys. Returns all storage keys of the given address
+func (api *ParityAPIImpl) ListStorageKeys(ctx context.Context, account common.Address, quantity int, offset *hexutil.Bytes) ([]hexutil.Bytes, error) {
+	tx, txErr := api.db.BeginRo(ctx)
+	if txErr != nil {
+		return nil, fmt.Errorf("listStorageKeys cannot open tx: %w", txErr)
+	}
+	defer tx.Rollback()
+	a, err := state.NewPlainStateReader(tx).ReadAccountData(account)
+	if err != nil {
+		return nil, err
+	} else if a == nil {
+		return nil, fmt.Errorf("acc not found")
+	}
+	c, err := tx.Cursor(kv.PlainState)
+	if err != nil {
+		return nil, err
+	}
+	keys := make([]hexutil.Bytes, 0)
+	var (
+		k []byte
+		e error
+	)
+	if offset != nil {
+		k, _, e = c.SeekExact(*offset)
```

Comment:
I don't understand what is Offset from docs https://openethereum.github.io/JSONRPC-parity-module#parity_liststoragekeys

Reply:
if it's "first storage to start from" then you may implement it with "DupSortCursor" - by method .SeekBothRange

Reply:
it's said that it is a storage key to start from, so I understood it as like we have keys
```
0x213
0x21344555
0x213467888
0x213467999
0x214567
0x21477
```
and offset of 0x21346 of account 0x213  (shortened the values) will give us 
```
0x213467888
0x213467999
```
but not sure about it

Reply:
as I see, SeekBothRange returns values, not keys